modules/xfem/src/base/XFEMFuncs.C

```diff
@@ -642,4 +642,141 @@ normalizePoint(EFAPoint & p)
     p *= (1.0 / len);
 }
 
+double
+r8_acos(double c)
+//****************************************************************************80
+//
+//  Purpose:
+//
+//    R8_ACOS computes the arc cosine function, with argument truncation.
+//
+//  Discussion:
+//
+//    If you call your system ACOS routine with an input argument that is
+//    outside the range [-1.0, 1.0 ], you may get an unpleasant surprise.
+//    This routine truncates arguments outside the range.
+//
+//  Licensing:
+//
+//    This code is distributed under the GNU LGPL license.
+//
+//  Modified:
+//
+//    13 June 2002
+//
+//  Author:
+//
+//    John Burkardt
+//
+//  Parameters:
+//
+//    Input, double C, the argument, the cosine of an angle.
+//
+//    Output, double R8_ACOS, an angle whose cosine is C.
+//
+{
+#define PI 3.141592653589793
```

Comment:
PI is defined in libMesh. It's also defined in some of these other stand alone XFEM utility files.

Reply:
This obviously here because this code was straight-up copied without modification from John Burkhardt. I'm totally on-board with using geometric routines in libmesh if they are available instead of these. Since this only adds one function to a large set of functions like this, my opinion is that it would be better to tackle replacing all of these functions as a separate task rather than figure out whether we have the needed functionality in libMesh and use that as part of this PR.  This PR has been in the review phase for a long time, and I'd like to get this code in now and have our team start working on cleaning a few things like that up rather than asking more of @liuusu here.